printer_mods/chirpy/NozzleWiper/nozzlewiper.cfg

```diff
@@ -0,0 +1,99 @@
+###############################################################################
+## NOZZLE WIPER - A nozzle purge bucket and brush for the Voron 0.1 and other
+##                printers for ants
+##
+## Configuration:
+##
+## 1.  Set the correct PMW pin that you connected the Orange wire to in the
+##     [servo wipeServo] section
+## 2.  Home X, Y and Z axes
+## 3.  Lower Z by at least 50mm
+## 4.  Run the NW_DEPLOY macro and ensure the servo extends the arm 90 degrees
+## 5.  Use the printer controls to move the nozzle so that it is inline with
+##     the brush on the X axis and so that it is in the centre of the purge
+##     bucket on the Y axis
+## 7.  Run M114 to get the nozzle position and set variable_x and variable_y
+## 8.  Move the nozzle into the center of the brush and raise the mount on
+##     the extrusion until the nozzle just rests into the brush. Do not let the
+##     brush touch the sock or it'll wear it out prematurely
+## 9.  Move the nozzle out of the way from the brush and purge bucket
+## 10. Run NW_RETRACT and ensure the servo moves the arm back away to the side
+##
+## Testing:
+##
+## 1.  Heat the nozzle to the printing temperature of the loaded filament
+## 2.  Home all axes
+## 3.  Be prepared to hit the Emergency Stop button if things don't go as they
+##     should
+## 4.  Run NW_CLEAN_NOZZLE
+##
+## That should be all changes needed. Call the NW_CLEAN_NOZZLE macro from
+## the end of your START_PRINT macro to wipe the nozzle clean before printing
+###############################################################################
+
+[servo wipeServo]
+pin: PA8
+minimum_pulse_width: 0.0005
+maximum_servo_angle: 180
+
+[gcode_macro NW_CLEAN_NOZZLE]
+gcode:
+   {% if "xyz" in printer.toolhead.homed_axes %}
+      SAVE_GCODE_STATE NAME=NW_CLEAN_NOZZLE
+      G90
+      NW_DEPLOY
+      NW_PURGE
+      NW_WIPE
+      NW_RETRACT
+      RESTORE_GCODE_STATE NAME=NW_CLEAN_NOZZLE
+   {% else %}
+      { action_raise_error("Home All Axis First") }
+      M117 Home All Axis First
+   {% endif %}
+
+[gcode_macro NW_BUCKET_POS]
+variable_x: 0
+variable_y: 90
```

Comment:
Is there any risk of damage to the printer for having default values here? Is there a way to use unrealistic values (-1000) to force the user to calculate the correct values on their own?

Reply:
There should not be any risk, especially if they follow the instructions. However, since that can easily be skipped, I will set the X/Y coordinates to negative numbers to force a fail if not set correctly.

Reply:
I've just pushed the requested changes.